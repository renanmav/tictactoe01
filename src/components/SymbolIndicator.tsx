import React from 'react';
import { Image, Text, View } from 'react-native';
import { t } from 'react-native-tailwindcss';

const s = {
  wrapper: [t.wFull, t.flexRow],
  container: [t.w3_12, t.itemsCenter, t.justifyCenter],
  fill: [t.flexGrow],
  text: [t.textWhite, t.textXl, t.fontBold],
  symbol: [t.objectContain, t.mB5, { height: 50, width: 50 }],
};

const SymbolIndicator: React.FC = () => {
  return (
    <View style={s.wrapper}>
      <View style={s.container}>
        <Image source={require('../../assets/x.png')} style={s.symbol} />
        <Text style={s.text}>VocÃª</Text>
      </View>
      <View style={s.fill} />
      <View style={s.container}>
        <Image source={require('../../assets/o.png')} style={s.symbol} />
        <Text style={s.text}>Bot</Text>
      </View>
    </View>
  );
};

export default SymbolIndicator;
