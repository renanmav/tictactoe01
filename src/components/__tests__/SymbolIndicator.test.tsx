import React from 'react';
import { render } from '@testing-library/react-native';

import SymbolIndicator from '../SymbolIndicator';

it('should render correctly', async () => {
  const { baseElement } = render(<SymbolIndicator />);

  expect(baseElement).toMatchSnapshot();
});
