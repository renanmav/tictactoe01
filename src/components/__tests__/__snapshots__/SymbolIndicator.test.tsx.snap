// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "resizeMode": "contain",
              },
              Object {
                "marginBottom": 20,
              },
              Object {
                "height": 50,
                "width": 50,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
              },
              Object {
                "fontSize": 20,
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          VocÃª
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexGrow": 1,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "width": "25%",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "resizeMode": "contain",
              },
              Object {
                "marginBottom": 20,
              },
              Object {
                "height": 50,
                "width": 50,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
              },
              Object {
                "fontSize": 20,
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          Bot
        </Text>
      </View>
    </View>
  </View>
</View>
`;
